packages/compass-preferences-model/src/preferences.ts

```diff
@@ -917,30 +788,30 @@ export class Preferences {
     attributes: Partial<UserPreferences> = {}
   ): Promise<AllPreferences> {
     const keys = Object.keys(attributes) as (keyof UserPreferences)[];
-    const originalPreferences = await this.fetchPreferences();
+    const originalPreferences = this.getPreferences();
     if (keys.length === 0) {
       return originalPreferences;
     }
 
-    const invalidKey = keys.find((key) => !modelPreferencesProps[key]);
+    const invalidKey = keys.find((key) => !storedUserPreferencesProps[key]);
     if (invalidKey !== undefined) {
       // Guard against accidentally saving non-model settings here.
       throw new Error(
         `Setting "${invalidKey}" is not part of the preferences model`
       );
     }
 
-    const userPreferencesModel = this._userPreferencesModel;
-
-    // Save user preferences to the Ampersand model.
-    const saveUserPreferences: (
-      attributes: Partial<UserPreferences>
-    ) => Promise<void> = promisifyAmpersandMethod(
-      userPreferencesModel.save.bind(userPreferencesModel)
-    );
+    keys.forEach((key) => {
+      const { error } = storedUserPreferencesProps[key].validator.validate(
+        attributes[key]
+      );
+      if (error) {
+        throw error;
+      }
+    });
```

Comment:
for-of should work fine, if you want:

```suggestion
    for (const key of keys) {
      const { error } = storedUserPreferencesProps[key].validator.validate(
        attributes[key]
      );
      if (error) {
        throw error;
      }
    }
```